### PR TITLE
feat(tests): Changes to fix tests and make sure we can run them successfully.

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        public DevicesClientTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [Test]
+        public async Task Devices_Lifecycle()
+        {
+            // TODO: This is just a verification that tests run and it requires the tester to complete this test however they see fit.
+            string testDeviceName = GetRandom();
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs:22]: Maybe add a suffix to this?

File: sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+            IoTHubServiceClient client = GetClient();
+
+            try
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs:26]: Just to set a pattern, we try to follow the pattern to add a try catch around the whole test and clean up in the end even if test failed mid way. And probably assert if cleanup fails. I know this is just a placeholder but probably adding a skeleton will help others write tests consistently.

**Review by vinagesh — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs (hunk 1 above)
Comment (azabbasi) [reply sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs:22]: Ok

**Review by barustum — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            : base(isAsync)
+        {
+        }
+
+        [Test]
+        [Category("Live")]
+        public async Task Devices_Lifecycle()
+        {
+            // TODO: This is just a verification that tests run and it requires the tester to complete this test however they see fit.
+            string testDeviceName = $"testDevice{GetRandom()}";
```
Comment (drwill-ms) [inline sdk/iot/Azure.Iot.Hub.Service/tests/DevicesClientTests.cs:25]: preferably the name of the test as a prefix. That way, it is easier to track back a device to the test that created it.

**Pull request merged**